pkg/setting/setting_feature_toggles.go

```diff
@@ -10,65 +13,188 @@ import (
 	"gopkg.in/ini.v1"
 )
 
+type FeatureToggleState string
+
+const (
+	AlphaState   FeatureToggleState = "alpha"   // anything can change without warning
+	BetaState    FeatureToggleState = "beta"    // shape looks right, but still evoloving
+	StableState  FeatureToggleState = "stable"  // stable and may soon become a build-in feature
+	MergedState  FeatureToggleState = "merged"  // feature no longer requires an explicit flag
+	RemovedState FeatureToggleState = "removed" // the feature was not merged and no longer has any effect
+)
+
 var (
 	featureToggleInfo = promauto.NewGaugeVec(prometheus.GaugeOpts{
 		Name:      "feature_toggles_info",
 		Help:      "info metric that exposes what feature toggles are enabled or not",
 		Namespace: "grafana",
 	}, []string{"name"})
+)
+
+type FeatureToggleInfo struct {
+	Id          string             `json:"id"`   // the key listed in grafana.ini
+	Name        string             `json:"name"` // user
+	Description string             `json:"description"`
+	AliasIds    []string           `json:"aliasIds,omitempty"` // ids that will also work (support spelling changes and product renames)
+	State       FeatureToggleState `json:"state,omitempty"`
+	DocsURL     string             `json:"docsURL,omitempty"`
+
+	// Loading state
+	Enabled bool `json:"enabled,omitempty"`
+
+	// Special flags
+	RequiresDevMode    bool `json:"requiresDevMode,omitempty"`    // can not be enabled in production
+	RequiresEnterprise bool `json:"requiresEnterprise,omitempty"` // requires enterprise executable
+	ModifiesDatabase   bool `json:"modifiesDatabase,omitempty"`   // the feature will modify the SQL tables
```

Comment:
I think this should be part of the state instead. Once the feature is mature enough to be tested in production we have to avoid breaking changes to the schema.

Reply:
ie: state=alpha-DB?

FYI: I actually started this effort when starting a feature that adds a SQL table, and was looking for a good way to shout that.  My assumption is that if it got merged to main it is reasonable if future "migration" actually drops all data first :grimacing: